### PR TITLE
Zonda fetch ticker v2

### PR DESCRIPTION
**Pull request by samgermain**
fixes: #13921

### v1_01PublicGetTradingTicker (Default)

```
% zonda fetchTicker BTC/USDT              
(node:10895) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
2023-06-21T03:31:39.311Z
Node.js: v18.15.0
CCXT v3.1.45
zonda.fetchTicker (BTC/USDT)
2023-06-21T03:31:41.150Z iteration 0 passed in 1175 ms

{
  symbol: 'BTC/USDT',
  timestamp: 1687318299476,
  datetime: '2023-06-21T03:31:39.476Z',
  high: undefined,
  low: undefined,
  bid: 28634.800006,
  bidVolume: undefined,
  ask: 28700,
  askVolume: undefined,
  vwap: undefined,
  open: undefined,
  close: 28634.800005,
  last: 28634.800005,
  previousClose: 28745,
  change: undefined,
  percentage: undefined,
  average: undefined,
  baseVolume: undefined,
  quoteVolume: undefined,
  info: {
    market: {
      code: 'BTC-USDT',
      first: { currency: 'BTC', minOffer: '0.0000038', scale: '8' },
      second: { currency: 'USDT', minOffer: '0.108', scale: '6' },
      amountPrecision: '8',
      pricePrecision: '6',
      ratePrecision: '6'
    },
    time: '1687318299476',
    highestBid: '28634.800006',
    lowestAsk: '28700',
    rate: '28634.800005',
    previousRate: '28745'
  }
}
2023-06-21T03:31:41.150Z iteration 1 passed in 1175 ms
```

```
2023-06-21T03:30:53.860Z
Node.js: v18.15.0
CCXT v3.1.45
zonda.fetchTickers (BTC/USDT,ETH/USDT)
2023-06-21T03:30:55.811Z iteration 0 passed in 1191 ms

{
  'ETH/USDT': {
    symbol: 'ETH/USDT',
    timestamp: 1687318215277,
    datetime: '2023-06-21T03:30:15.277Z',
    high: undefined,
    low: undefined,
    bid: 1802.82,
    bidVolume: undefined,
    ask: 1808.46,
    askVolume: undefined,
    vwap: undefined,
    open: undefined,
    close: 1803.840002,
    last: 1803.840002,
    previousClose: 1803.12,
    change: undefined,
    percentage: undefined,
    average: undefined,
    baseVolume: undefined,
    quoteVolume: undefined,
    info: {
      market: {
        code: 'ETH-USDT',
        first: { currency: 'ETH', minOffer: '0.00006', scale: '8' },
        second: { currency: 'USDT', minOffer: '0.108', scale: '6' },
        amountPrecision: '8',
        pricePrecision: '6',
        ratePrecision: '6'
      },
      time: '1687318215277',
      highestBid: '1802.82',
      lowestAsk: '1808.46',
      rate: '1803.840002',
      previousRate: '1803.12'
    }
  },
  'BTC/USDT': {
    symbol: 'BTC/USDT',
    timestamp: 1687318251087,
    datetime: '2023-06-21T03:30:51.087Z',
    high: undefined,
    low: undefined,
    bid: 28634.800003,
    bidVolume: undefined,
    ask: 28737.289998,
    askVolume: undefined,
    vwap: undefined,
    open: undefined,
    close: 28745,
    last: 28745,
    previousClose: 28746.7,
    change: undefined,
    percentage: undefined,
    average: undefined,
    baseVolume: undefined,
    quoteVolume: undefined,
    info: {
      market: {
        code: 'BTC-USDT',
        first: { currency: 'BTC', minOffer: '0.0000038', scale: '8' },
        second: { currency: 'USDT', minOffer: '0.108', scale: '6' },
        amountPrecision: '8',
        pricePrecision: '6',
        ratePrecision: '6'
      },
      time: '1687318251087',
      highestBid: '28634.800003',
      lowestAsk: '28737.289998',
      rate: '28745',
      previousRate: '28746.7'
    }
  }
}
2023-06-21T03:30:55.811Z iteration 1 passed in 1191 ms
```

```
2023-06-21T03:28:03.496Z
Node.js: v18.15.0
CCXT v3.1.45
zonda.fetchTickers ()
2023-06-21T03:28:05.404Z iteration 0 passed in 1213 ms

{
  'BTG/EUR': {
    symbol: 'BTG/EUR',
    timestamp: 1687228547727,
    datetime: '2023-06-20T02:35:47.727Z',
    high: undefined,
    low: undefined,
    bid: 9.6,
    bidVolume: undefined,
    ask: 12.93,
    askVolume: undefined,
    vwap: undefined,
    open: undefined,
    close: 10.01,
    last: 10.01,
    previousClose: 11.72,
    change: undefined,
    percentage: undefined,
    average: undefined,
    baseVolume: undefined,
    quoteVolume: undefined,
    info: {
      ...
    }
  },
  ...
  2023-06-21T03:28:05.404Z iteration 1 passed in 1213 ms
```

### v1_01PublicGetTradingStats

```
% zonda fetchTicker BTC/USDT '{"method": "v1_01PublicGetTradingStatsSymbol"}'          
2023-06-21T03:40:06.104Z
Node.js: v18.15.0
CCXT v3.1.45
zonda.fetchTicker (BTC/USDT, [object Object])
2023-06-21T03:40:07.943Z iteration 0 passed in 1186 ms

{
  symbol: 'BTC/USDT',
  timestamp: undefined,
  datetime: undefined,
  high: 28800,
  low: 26703.950101,
  bid: undefined,
  bidVolume: undefined,
  ask: undefined,
  askVolume: undefined,
  vwap: undefined,
  open: 27122.2,
  close: undefined,
  last: undefined,
  previousClose: undefined,
  change: undefined,
  percentage: undefined,
  average: undefined,
  baseVolume: 6.72932396,
  quoteVolume: undefined,
  info: {
    m: 'BTC-USDT',
    h: '28800',
    l: '26703.950101',
    v: '6.72932396',
    r24h: '27122.2'
  }
}
2023-06-21T03:40:07.943Z iteration 1 passed in 1186 ms
```

```
2023-06-21T03:41:36.728Z
Node.js: v18.15.0
CCXT v3.1.45
zonda.fetchTickers (, [object Object])
2023-06-21T03:41:38.583Z iteration 0 passed in 1199 ms

{
  'BTG/EUR': {
    symbol: 'BTG/EUR',
    timestamp: undefined,
    datetime: undefined,
    high: undefined,
    low: undefined,
    bid: undefined,
    bidVolume: undefined,
    ask: undefined,
    askVolume: undefined,
    vwap: undefined,
    open: 10.01,
    close: undefined,
    last: undefined,
    previousClose: undefined,
    change: undefined,
    percentage: undefined,
    average: undefined,
    baseVolume: 0,
    quoteVolume: undefined,
    info: { m: 'BTG-EUR', h: null, l: null, v: '0', r24h: '10.01' }
  },
...
```

### CODE REVIEW

File: ts/src/zonda.ts [hunk 1]
```diff
@@ -244,6 +244,8 @@ export default class zonda extends Exchange {
                 },
             },
             'options': {
+                'fetchTickerMethod': 'fetchTickerV2', // fetchTickerV1, fetchTickerV2
```
Comment (pcriadoperez) [inline ts/src/zonda.ts:247]: I got a bit confused with the v1 and v2, expecting it to be the api version.
Would it make sense to call them pass the endpoint method here: `v1_01PublicGetTradingTickerSymbol` or `v1_01PublicGetTradingStatsSymbol`?
Or call them `ticket` and `stats`

File: ts/src/zonda.ts [hunk 2]
```diff
@@ -633,6 +661,25 @@ export default class zonda extends Exchange {
          * @name zonda#fetchTicker
          * @see https://docs.zonda.exchange/reference/market-statistics
          * @description fetches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @param {str} symbol unified symbol of the market to fetch the ticker for
```
Comment (pcriadoperez) [inline ts/src/zonda.ts:664]:
```suggestion
         * @param {string} symbol unified symbol of the market to fetch the ticker for
```

File: ts/src/zonda.ts [hunk 3]
```diff
@@ -633,6 +661,25 @@ export default class zonda extends Exchange {
...
+         * @param {dict} params extra parameters specific to the gemini api endpoint
```
Comment (pcriadoperez) [inline ts/src/zonda.ts:665]:
```suggestion
         * @param {object} params extra parameters specific to the gemini api endpoint
```

File: ts/src/zonda.ts [hunk 4]
```diff
@@ -633,6 +661,25 @@ export default class zonda extends Exchange {
...
+         * @param {dict} params.fetchTickerMethod 'fetchTickerV2' or 'fetchTickerV1' - 'fetchTickerV1' for original ccxt.zonda.fetchTicker - uses this.options['fetchTickerMethod'] if not set - default = 'fetchTickerV2'
+         * @returns {dict} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const defaultMethod = this.safeString (this.options, 'fetchTickerMethod', 'fetchTickerV2');
+        const fetchTickerMethod = this.safeString (params, 'fetchTickerMethod', defaultMethod);
+        if (fetchTickerMethod === 'fetchTickerV1') {
```
Comment (pcriadoperez) [inline ts/src/zonda.ts:671]: fetchTickerV1 and fetchTickerV2 seem the same with the only difference that they call a different method. Could this be simplified to do something like this instead?
```
await this.loadMarkets ();
        const market = this.market (symbol);
        const request = {
            'symbol': market['id'],
        };
        const response = await this.[method] (this.extend (request, params));
        const rawTicker = this.safeValue2 (response, 'stats', 'items');
        return this.parseTicker (rawTicker);
        }
        ```

Comment (samgermain) [reply ts/src/zonda.ts:671]: @carlosmiei doesn't like the dynamic method calls

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**